### PR TITLE
Render Tooltip in a portal to avoid it being cut off

### LINKED ISSUE #1528: Tooltip is cut off when used in a custom cell of data grid
### Package name(s)

Core v1.4.0 (@salt-ds/core), Data Grid v1.0.3 (@salt-ds/data-grid)

### Description

When using Tooltip in a custom cell of data grid, tooltip is cut off within each individual cell.

### Steps to reproduce

https://stackblitz.com/edit/salt-grid-template-gjunhl?file=App.tsx

### Expected behavior

Tooltip display without obstruction

### Operating system

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

### PR DESCRIPTION
**Pull request by joshwooding**
fixes #1528

### CODE REVIEW

File: packages/core/src/tooltip/useTooltip.ts [hunk 1]
```diff
@@ -108,8 +108,9 @@ export function useTooltip(props?: UseTooltipProps) {
     ref: arrowRef,
     context,
     style: {
-      left: arrowX ? `${arrowX}px` : "",
-      top: arrowY ? `${arrowY}px` : "",
+      position: strategy,
+      left: arrowX ?? 0,
```
Comment (joshwooding) [inline packages/core/src/tooltip/useTooltip.ts:112]: This was needed to fix the positioning of Tooltips within the Portal. If not the arrow and body would be weird on the first render

File: packages/core/src/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -99,42 +106,52 @@ export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
...
-                strokeWidth={1}
-                fill="var(--salt-container-primary-background)"
-                stroke="var(--tooltip-status-borderColor)"
-                height={5}
-                width={10}
-              />
-            )}
-          </div>
+          <FloatingPortal>
+            <SaltProvider>
```
Comment (joshwooding) [inline packages/core/src/tooltip/Tooltip.tsx:121]: The provider was needed to support the use case where an app has nested modes. The element that is portalled needs to have the same style as the current scope

File: packages/core/src/tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -99,42 +106,52 @@ export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
...
     );
 
+    const floatingRef = useForkRef(floating, ref) as Ref<HTMLDivElement>;
+
     return (
       <>
         {isValidElement(children) &&
           cloneElement(children, {
-            ...getTriggerProps(),
+            ...mergeProps(getTriggerProps(), children.props),
```
Comment (joshwooding) [inline packages/core/src/tooltip/Tooltip.tsx:115]: mergeProps is needed to merge the callbacks

File: .changeset/sweet-shrimps-allow.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+"@salt-ds/core": minor
+---
+
+Add mergeProps to utils.
+
+This utility merges two prop objects according to the following rules:
```
Comment (origami-z) [inline .changeset/sweet-shrimps-allow.md:7]: The detailed description feels more fit to either the doc site, or JsDoc comments.

File: packages/core/src/tooltip/Tooltip.tsx (hunk 2 above)
Comment (origami-z) [reply packages/core/src/tooltip/Tooltip.tsx:121]: Could these be inline comments? Sounds like good long term 'requirement' type of feature

**Review by origami-z — COMMENTED** [review]
Do we have anything to test the 'fix' of cutoff? maybe storybook?

**Review by origami-z — APPROVED** [review]

**Review by alycrys — APPROVED** [review]

**Pull request merged**